packages/api-server/src/methods/modules/eth.ts

```diff
@@ -724,46 +731,74 @@ export class Eth {
     if (tx != null) {
+      // no need await
+      // delete auto create account tx if already in db
+      this.cacheStore.delete(cacheKey);
       const apiTx = toApiTransaction(tx);
       return apiTx;
     }
 
-    // if null, find pending transactions
-    const godwokenTxWithStatus = await this.rpc.getTransaction(gwTxHash);
-    if (godwokenTxWithStatus == null) {
-      return null;
-    }
-    const godwokenTxReceipt = await this.rpc.getTransactionReceipt(gwTxHash);
-    const tipBlock = await this.query.getTipBlock();
-    if (tipBlock == null) {
-      throw new Error("tip block not found!");
-    }
-    let ethTxInfo = undefined;
-    try {
-      ethTxInfo = await filterWeb3Transaction(
-        ethTxHash,
-        this.rpc,
-        tipBlock.number,
-        tipBlock.hash,
-        godwokenTxWithStatus.transaction,
-        godwokenTxReceipt
-      );
-    } catch (err) {
-      logger.error("filterWeb3Transaction:", err);
-      logger.info("godwoken tx:", godwokenTxWithStatus);
-      logger.info("godwoken receipt:", godwokenTxReceipt);
-      throw err;
+    // 2. If null, find pending transactions
+    const ethTxHashKey = ethTxHashCacheKey(ethTxHash);
+    const gwTxHash: Hash | null = await this.cacheStore.get(ethTxHashKey);
+    if (gwTxHash != null) {
+      const godwokenTxWithStatus = await this.rpc.getTransaction(gwTxHash);
+      if (godwokenTxWithStatus == null) {
+        return null;
+      }
+      const godwokenTxReceipt = await this.rpc.getTransactionReceipt(gwTxHash);
+      const tipBlock = await this.query.getTipBlock();
+      if (tipBlock == null) {
+        throw new Error("tip block not found!");
+      }
+      let ethTxInfo = undefined;
+      try {
+        ethTxInfo = await filterWeb3Transaction(
+          ethTxHash,
+          this.rpc,
+          tipBlock.number,
+          tipBlock.hash,
+          godwokenTxWithStatus.transaction,
+          godwokenTxReceipt
+        );
+      } catch (err) {
+        logger.error("filterWeb3Transaction:", err);
+        logger.info("godwoken tx:", godwokenTxWithStatus);
+        logger.info("godwoken receipt:", godwokenTxReceipt);
+        throw err;
+      }
+      if (ethTxInfo != null) {
+        const ethTx = ethTxInfo[0];
+        return ethTx;
+      }
     }
-    if (ethTxInfo != null) {
-      const ethTx = ethTxInfo[0];
-      return ethTx;
+
+    // 3. Find by auto create account tx
+    // TODO: delete cache store if dropped by godwoken
+    // convert to tx hash mapping store if account id generated ?
+    const polyjuiceRawTx = await this.cacheStore.get(cacheKey);
+    if (polyjuiceRawTx != null) {
+      const tipBlock = await this.query.getTipBlock();
+      if (tipBlock == null) {
+        throw new Error("tip block not found!");
+      }
+      // Convert polyjuice tx to api transaction
+      const { tx, fromAddress }: AutoCreateAccountCacheValue =
+        JSON.parse(polyjuiceRawTx);
+      const apiTransaction: EthTransaction =
```

Comment:
Why is an `EthTransaction` called `apiTransaction`?
- https://github.com/nervosnetwork/godwoken-web3/pull/430/files#diff-cce75f92de0f01f77430b0195c78d8a547ca11f2b8f818202008dce92a69ebefR793
- https://github.com/nervosnetwork/godwoken-web3/pull/430/files#diff-cce75f92de0f01f77430b0195c78d8a547ca11f2b8f818202008dce92a69ebefR742

Reply:
For RPC response format